controllers/routing/exported_svc_locator.go

```diff
@@ -0,0 +1,38 @@
+package routing
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/opendatahub-io/odh-platform/pkg/metadata"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func getExportedService(ctx context.Context, cli client.Client, target *unstructured.Unstructured) (*corev1.Service, error) {
+	exportedSvcList := &corev1.ServiceList{}
+	listOpts := []client.ListOption{
+		client.InNamespace(target.GetNamespace()),
+		client.MatchingLabels{metadata.Labels.RoutingExported: "true"},
+	}
+
+	if errList := cli.List(ctx, exportedSvcList, listOpts...); errList != nil {
+		return nil, fmt.Errorf("could not list exported services: %w", errList)
+	}
+
+	if len(exportedSvcList.Items) == 0 {
+		return nil, &NoExportedServicesError{Target: target}
+	}
+
+	return &exportedSvcList.Items[0], nil
```

Comment:
So we are (for now) limiting to one exported service per namespace?

Reply:
Good catch. We can easily handle more.

Reply:
30c0a49